File: app/src/main/java/za/co/jpsoft/winkerkreader/data/FilterCheckBox.java
package za.co.jpsoft.winkerkreader.data;

import java.io.Serializable;
import java.util.List;

/**
 * Created by Pieter Grobler on 06/09/2017.
 */

public class FilterCheckBox implements Serializable {

    private static final long serialVersionUID = 1L;

    private static final String DEFAULT_DESCRIPTION = "N/A";

    private String ftitle = "";
    private List <String> flist1 = null;
    private String ftext1 = "";
    private String ftext2 = "";
    private String ftext3 = "";
    private int fcount = 1;
    private boolean checked = false;

// get functions
    public String getTitle () {
        return ftitle == null ? DEFAULT_DESCRIPTION  : ftitle;
    }
    public String getText1() {
        return ftext1 == null ? DEFAULT_DESCRIPTION  : ftext1;
    }
    public String getText2 () {
        return ftext2 == null ? DEFAULT_DESCRIPTION  : ftext2;
    }
    public String getText3 () {
        return ftext3 == null ? DEFAULT_DESCRIPTION : ftext3;
    }
    public int getCount () {
        return fcount;
    }
    public List getList() {
        return flist1 == null ? null  : flist1;
    }
    public boolean getChecked () {
        return checked;
    }

//Set functions
    public void setTitle(String detail) { this.ftitle = detail; }
    public void setText1(String detail) { this.ftext1 = detail; }
    public void setText2(String detail) { this.ftext2 = detail; }
    public void setText3(String detail) { this.ftext3 = detail; }
    public void setCount(int detail) { this.fcount = detail; }
    public void setList(List<String> list) { this.flist1 = list; }
    public void setChecked ( final boolean checked ) {
        this.checked = checked;
    }



    public FilterCheckBox (final String title, final List <String> list1, final String text1, final String text2, final String text3, final int count) {
        this.ftitle = title;
        this.flist1 = list1;
        this.ftext1 = text1;
        this.ftext2 = text2;
        this.ftext3 = text3;
        this.fcount = count;

    }

}

File: app/src/main/kotlin/FilterCheckBox.kt
package za.co.jpsoft.winkerkreader.data

import android.os.Parcel
import android.os.Parcelable
import java.io.Serializable

/**
 * Created by Pieter Grobler on 06/09/2017.
 */
class FilterCheckBox @JvmOverloads constructor(
    title: String = "",
    list: List<String>? = null,
    text1: String = "",
    text2: String = "",
    text3: String = "",
    count: Int = 1,
    checked: Boolean = false
) : Serializable {

    companion object {
        private const val DEFAULT_DESCRIPTION = "N/A"
    }

    private var _title: String = title
    private var _list: List<String>? = list
    private var _text1: String = text1
    private var _text2: String = text2
    private var _text3: String = text3
    private var _count: Int = count
    private var _checked: Boolean = checked

    var title: String
        get() = _title.ifEmpty { DEFAULT_DESCRIPTION }
        set(value) { _title = value }

    var list: List<String>?
        get() = _list
        set(value) { _list = value }

    var text1: String
        get() = _text1.ifEmpty { DEFAULT_DESCRIPTION }
        set(value) { _text1 = value }

    var text2: String
        get() = _text2.ifEmpty { DEFAULT_DESCRIPTION }
        set(value) { _text2 = value }

    var text3: String
        get() = _text3.ifEmpty { DEFAULT_DESCRIPTION }
        set(value) { _text3 = value }

    var count: Int
        get() = _count
        set(value) { _count = value }

    var checked: Boolean
        get() = _checked
        set(value) { _checked = value }
}